File: Cpp/2405.cpp
class Solution {
public:
    int partitionString(string s) {
        set<char> ss;
        int ctr = 0;
        for(int i = 0; i < s.length(); i++){
            if(ss.find(s[i]) == ss.end()){
                ss.insert(s[i]);
            }else{
                ss.clear();
                ss.insert(s[i]);
                ctr++;
            }
        }
        ctr++;
        return ctr;
        
    }
};

File: Python/2405.py
class Solution:
    def partitionString(self, s: str) -> int:
        ss = set()
        sol = 0
        for i in range(len(s)):
            if s[i] in ss:
                ss.clear()
                sol += 1
            ss.add(s[i])
        return sol+1
